pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -133,7 +133,7 @@ let
       installPhase = ''
         export HOME="$TMP"
         mkdir -p "$out"
-        cabal2nix --compiler=${self.ghc.haskellCompilerName} --system=${hostPlatform.config} ${sha256Arg} "${src}" ${extraCabal2nixOptions} > "$out/default.nix"
+        cabal2nix --compiler=${self.ghc.haskellCompilerName} ${stdenv.lib.optionalString (!hostPlatform.isWasm) "--system=${hostPlatform.config}"} ${sha256Arg} "${src}" ${extraCabal2nixOptions} > "$out/default.nix"
```

Comment:
I suppose this is fine for now, but can you add a "TODO patch Cabal?"

Reply:
Got rid of this, on the latest nixpkgs things work fine without this change.